drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1218 @@
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	uint8_t regdata = 0;
+	int retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_SENSOR_CONFIG_2,
+		&regdata);
+
+	if (retval != 0) {
+		return retval;
+	}
+
+	val->val1 = (regdata & TMAG5273_ANGLE_EN_MSK) >> TMAG5273_ANGLE_EN_POS;
+
+	return 0;
+}
+
+/* -------------------------------------------------------------------------------------- */
+
+static int tmag5273_attr_set(const struct device *dev,
+				enum sensor_channel chan,
+				enum sensor_attribute attr,
+				const struct sensor_value *val)
+{
+	if (dev == NULL) {
+		LOG_DBG("dev: NULL");
+		return -EINVAL;
+	}
+
+	if (val == NULL) {
+		LOG_DBG("val: NULL");
+		return -EINVAL;
+	}
+
+	int retval = 0;
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	retval = -ENOTSUP;
+
+	switch ((uint16_t)attr) {
+	case TMAG5273_MEAS_RANGE: {
+		if (drv_cfg->meas_range == TMAG5273_AXIS_RANGE_RUNTIME) {
+			retval = tmag5273_attr_set_xyz_meas_range(dev, val);
+		}
+
+		break;
+	}
+	case TMAG5273_ATTR_CALC_AXIS: {
+		if (drv_cfg->angle_magnitude_axis == TMAG5273_ANGLE_MAG_RUNTIME) {
+			retval = tmag5273_attr_set_xyz_calc(dev, val);
+		}
+		break;
+	}
+	default: {
+		break;
+	}
+	}
+
+	if (retval) {
+		LOG_DBG("unknown attribute |%d|", attr);
+	}
+
+	return retval;
+}
+
+static int tmag5273_attr_get(const struct device *dev,
+		enum sensor_channel chan,
+		enum sensor_attribute attr,
+		struct sensor_value *val)
+{
+	if (dev == NULL) {
+		LOG_DBG("dev: NULL");
+		return -EINVAL;
+	}
+
+	if (val == NULL) {
+		LOG_DBG("val: NULL");
+		return -EINVAL;
+	}
+
+	int retval = 0;
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	retval = -ENOTSUP;
+
+	switch ((uint16_t)attr) {
+	case TMAG5273_MEAS_RANGE: {
+		if (drv_cfg->meas_range == TMAG5273_AXIS_RANGE_RUNTIME) {
+			retval = tmag5273_attr_get_xyz_meas_range(dev, val);
+		}
+
+		break;
+	}
+	case TMAG5273_ATTR_CALC_AXIS: {
+		if (drv_cfg->angle_magnitude_axis ==  TMAG5273_ANGLE_MAG_RUNTIME) {
+			retval = tmag5273_attr_get_xyz_calc(dev, val);
+		}
+
+		break;
+	}
+	default: {
+		LOG_DBG("unknown attribute |%d|", attr);
+		break;
+	}
+	}
+
+	return retval;
+}
+
+static int tmag5273_sample_fetch(const struct device *dev,
+		enum sensor_channel chan)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	int retval = 0;
+
+	/* trigger measurement if in standby-mode */
+
+	/* trigger a conversion and wait until done if in standby mode */
+	if (drv_cfg->operation_mode == TMAG5273_OPER_MODE_STANDBY) {
+		if (drv_cfg->trigger_conv_via_int) {
+			retval = _tmag5273_dev_int_trigger(dev);
+
+			if (retval < 0) {
+				return retval;
+			}
+		}
+
+		drv_data->i2c_buffer[0] = TMAG5273_SET_REG_BITS(0x00, TMAG5273_RESULT_STATUS,
```

Comment:
Could you explain the benefits to me? I'm just genuinely interested in the pros doing it instead of keeping the memory chunk in the driver data structure.